diff --git a/aloja-web/Controller/MLCrossvarController.php b/aloja-web/Controller/MLCrossvarController.php
index 0518dcd..f295142 100644
--- a/aloja-web/Controller/MLCrossvarController.php
+++ b/aloja-web/Controller/MLCrossvarController.php
@@ -33,13 +33,10 @@ class MLCrossvarController extends AbstractController
 			|| (count($_GET) == 2 && array_key_exists('current_model',$_GET)))
 			{
 				$where_configs = '';
-				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['benchs'] = array('wordcount'); $where_configs .= ' AND bench IN ("wordcount")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")'; 			
 			}
 			$where_configs = str_replace("AND .","AND ",$where_configs);
@@ -209,14 +206,11 @@ class MLCrossvarController extends AbstractController
 			|| (count($_GET) == 2 && array_key_exists('current_model',$_GET)))
 			{
 				$where_configs = '';
-				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['benchs'] = array('wordcount'); $where_configs .= ' AND bench IN ("wordcount")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
-				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")'; 		
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
+				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")'; 			
 			}
 			$where_configs = str_replace("AND .","AND ",$where_configs);
 			$where_configs = str_replace("id_cluster","e.id_cluster",$where_configs);
@@ -402,14 +396,14 @@ class MLCrossvarController extends AbstractController
 			|| (count($_GET) == 2 && array_key_exists('current_model',$_GET)))
 			{
 				$params['benchs'] = $_GET['benchs'] = array('terasort'); $where_configs = ' AND bench IN ("terasort")';
-				$params['disks'] = $_GET['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['disks'] = $_GET['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = $_GET['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = $_GET['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
+				$params['comps'] = $_GET['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = $_GET['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
 				$params['id_clusters'] = $_GET['id_clusters'] = array('1'); $where_configs .= ' AND id_cluster IN ("1")';
-				$params['mapss'] = $_GET['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = $_GET['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = $_GET['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
+				//$params['mapss'] = $_GET['mapss'] = array('4'); $where_configs .= ' AND maps IN ("4")';
+				//$params['iosfs'] = $_GET['iosfs'] = array('10'); $where_configs .= ' AND iosf IN ("10")';
+				$params['blk_sizes'] = $_GET['blk_sizes'] = array('128'); $where_configs .= ' AND blk_size IN ("128")';
 			}
 			$where_configs = str_replace("AND .","AND ",$where_configs);
 
diff --git a/aloja-web/Controller/MLFindAttributesController.php b/aloja-web/Controller/MLFindAttributesController.php
index a9f8af1..4a514e7 100644
--- a/aloja-web/Controller/MLFindAttributesController.php
+++ b/aloja-web/Controller/MLFindAttributesController.php
@@ -43,14 +43,14 @@ class MLFindAttributesController extends AbstractController
 			{
 				$where_configs = '';
 				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = array('65536','131072'); $where_configs .= ' AND iofilebuf IN ("65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
 				$params['id_clusters'] = array('1'); $where_configs .= ' AND id_cluster IN ("1")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
+				$params['mapss'] = array('4'); $where_configs .= ' AND maps IN ("4")';
+				$params['iosfs'] = array('10'); $where_configs .= ' AND iosf IN ("10")';
+				$params['blk_sizes'] = array('128'); $where_configs .= ' AND blk_size IN ("128")';
 				$unseen = FALSE;
 			}
 
diff --git a/aloja-web/Controller/MLMinconfigsController.php b/aloja-web/Controller/MLMinconfigsController.php
index c5ef96a..6842771 100644
--- a/aloja-web/Controller/MLMinconfigsController.php
+++ b/aloja-web/Controller/MLMinconfigsController.php
@@ -33,15 +33,12 @@ class MLMinconfigsController extends AbstractController
 			|| (count($_GET) == 2 && array_key_exists('learn',$_GET)))
 			{
 				$where_configs = '';
-				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';				
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
-				$unrestricted = TRUE;
+				$unrestricted = TRUE; 
 			}
 
 			// FIXME PATCH FOR PARAM LIBRARIES WITHOUT LEGACY
@@ -102,7 +99,7 @@ class MLMinconfigsController extends AbstractController
 				// run the R processor
 				exec('cd '.getcwd().'/cache/query; touch '.md5($config).'.lock');
 				$command = getcwd().'/resources/queue -c "cd '.getcwd().'/cache/query; ../../resources/aloja_cli.r -d '.$cache_ds.' -m '.$learn_method.' -p '.$learn_options.' >/dev/null 2>&1 && ';
-				$command = $command.'../../resources/aloja_cli.r -m aloja_minimal_instances -l '.md5($config).' -p saveall='.md5($config.'R').':kmax=10 >/dev/null 2>&1; rm -f '.md5($config).'.lock; touch '.md5($config).'.fin" >/dev/null 2>&1 &';
+				$command = $command.'../../resources/aloja_cli.r -m aloja_minimal_instances -l '.md5($config).' -p saveall='.md5($config.'R').':kmax=200 >/dev/null 2>&1; rm -f '.md5($config).'.lock; touch '.md5($config).'.fin" >/dev/null 2>&1 &';
 				exec($command);
 			}
 			$in_process = file_exists(getcwd().'/cache/query/'.md5($config).'.lock');
diff --git a/aloja-web/Controller/MLNewconfigsController.php b/aloja-web/Controller/MLNewconfigsController.php
index 5308270..88ce6af 100644
--- a/aloja-web/Controller/MLNewconfigsController.php
+++ b/aloja-web/Controller/MLNewconfigsController.php
@@ -35,12 +35,9 @@ class MLNewconfigsController extends AbstractController
 			{
 				$where_configs = '';
 				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
 				$params['datanodess'] = array('3');// $where_configs .= ' AND datanodes = 3';
 				$params['bench_types'] = array('HiBench');// $where_configs .= ' AND bench_type = "HiBench"';
@@ -111,7 +108,7 @@ class MLNewconfigsController extends AbstractController
 				$command = getcwd().'/resources/queue -c "cd '.getcwd().'/cache/query; ../../resources/aloja_cli.r -d '.$cache_ds.' -m '.$learn_method.' -p '.$learn_options.':saveall='.md5($config."F").':vin=\'Benchmark,Net,Disk,Maps,IO.SFac,Rep,IO.FBuf,Comp,Blk.size,Datanodes,Bench.Type,VM.OS,VM.Cores,VM.RAM,VM.Size,Hadoop.Version,Type\' >/dev/null 2>&1 && ';
 				$command = $command.'../../resources/aloja_cli.r -m aloja_predict_instance -l '.md5($config."F").' -p inst_predict=\''.$instance.'\':saveall='.md5($config."D").':vin=\'Benchmark,Net,Disk,Maps,IO.SFac,Rep,IO.FBuf,Comp,Blk.size,Datanodes,Bench.Type,VM.OS,VM.Cores,VM.RAM,VM.Size,Hadoop.Version,Type\' >/dev/null 2>&1 && ';
 				$command = $command.'../../resources/aloja_cli.r -d '.md5($config."D").'-dataset.data -m '.$learn_method.' -p '.$learn_options.':saveall='.md5($config."M").':vin=\'Benchmark,Net,Disk,Maps,IO.SFac,Rep,IO.FBuf,Comp,Blk.size,Datanodes,Bench.Type,VM.OS,VM.Cores,VM.RAM,VM.Size,Hadoop.Version,Type\' >/dev/null 2>&1 && ';
-				$command = $command.'../../resources/aloja_cli.r -m aloja_minimal_instances -l '.md5($config."M").' -p saveall='.md5($config.'R').':kmax=10 >/dev/null 2>&1; rm -f '.md5($config).'.lock; touch '.md5($config).'.fin" >/dev/null 2>&1 &';
+				$command = $command.'../../resources/aloja_cli.r -m aloja_minimal_instances -l '.md5($config."M").' -p saveall='.md5($config.'R').':kmax=200 >/dev/null 2>&1; rm -f '.md5($config).'.lock; touch '.md5($config).'.fin" >/dev/null 2>&1 &';
 				exec($command);
 			}
 			$in_process = file_exists(getcwd().'/cache/query/'.md5($config).'.lock');
diff --git a/aloja-web/Controller/MLOutliersController.php b/aloja-web/Controller/MLOutliersController.php
index 902c99e..33f1147 100644
--- a/aloja-web/Controller/MLOutliersController.php
+++ b/aloja-web/Controller/MLOutliersController.php
@@ -38,14 +38,15 @@ class MLOutliersController extends AbstractController
 			|| (count($_GET) == 3 && array_key_exists('dump',$_GET) && array_key_exists('current_model',$_GET)))
 			{
 				$where_configs = '';
-				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';				
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
-				$params['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
+				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
+				$params['iofilebufs'] = array('65536','131072'); $where_configs .= ' AND iofilebuf IN ("65536","131072")';
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
+//				$params['id_clusters'] = array('1'); $where_configs .= ' AND id_cluster IN ("1")';
+//				$params['mapss'] = array('4'); $where_configs .= ' AND maps IN ("4")';
+//				$params['iosfs'] = array('10'); $where_configs .= ' AND iosf IN ("10")';
+//				$params['blk_sizes'] = array('128'); $where_configs .= ' AND blk_size IN ("128")';
 			}
 
 			// FIXME PATCH FOR PARAM LIBRARIES WITHOUT LEGACY
diff --git a/aloja-web/Controller/MLParamevalController.php b/aloja-web/Controller/MLParamevalController.php
index a8a8b81..469e819 100644
--- a/aloja-web/Controller/MLParamevalController.php
+++ b/aloja-web/Controller/MLParamevalController.php
@@ -34,13 +34,12 @@ class MLParamevalController extends AbstractController
 			{
 				$params['benchs'] = $_GET['benchs'] = array('terasort'); $where_configs = ' AND bench IN ("terasort")';
 				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'net') $params['nets'] = $_GET['nets'] = array('ETH'); $where_configs .= ' AND net IN ("ETH")';
-				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'disk') $params['disks'] = $_GET['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
-				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'maps') $params['mapss'] = $_GET['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
+				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'disk') $params['disks'] = $_GET['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'iofilebuf') $params['iofilebufs'] = $_GET['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'iofs') $params['iosfs'] = $_GET['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'blk_size') $params['blk_sizes'] = $_GET['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
-				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'comp') $params['comps'] = $_GET['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
+				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'iofs') $params['iosfs'] = $_GET['iosfs'] = array('10'); $where_configs .= ' AND iosf IN ("10")';
+				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'comp') $params['comps'] = $_GET['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'replication') $params['replications'] = $_GET['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
+				if (!isset($_GET['parameval']) || $_GET['parameval'] != 'id_cluster') $params['id_clusters'] = $_GET['id_clusters'] = array('1','2','3'); $where_configs .= ' AND id_cluster IN ("1","2","3")';
 			}
 
 			$money		= Utils::read_params ( 'money', $where_configs, $configurations, $concat_config );
diff --git a/aloja-web/Controller/MLTemplatesController.php b/aloja-web/Controller/MLTemplatesController.php
index c00f108..3f52105 100644
--- a/aloja-web/Controller/MLTemplatesController.php
+++ b/aloja-web/Controller/MLTemplatesController.php
@@ -39,12 +39,9 @@ class MLTemplatesController extends AbstractController
  			{
 				$where_configs = '';
 				$params['benchs'] = array('terasort'); $where_configs .= ' AND bench IN ("terasort")';				
-				$params['disks'] = array('RR1'); $where_configs .= ' AND disk IN ("RR1")';
+				$params['disks'] = array('HDD','SSD'); $where_configs .= ' AND disk IN ("HDD","SSD")';
 				$params['iofilebufs'] = array('32768','65536','131072'); $where_configs .= ' AND iofilebuf IN ("32768","65536","131072")';
-				$params['comps'] = array('3'); $where_configs .= ' AND comp IN ("3")';
-				$params['mapss'] = array('2','4'); $where_configs .= ' AND maps IN ("2","4")';
-				$params['iosfs'] = array('100'); $where_configs .= ' AND iosf IN ("100")';
-				$params['blk_sizes'] = array('134'); $where_configs .= ' AND blk_size IN ("134")';
+				$params['comps'] = array('0'); $where_configs .= ' AND comp IN ("0")';
 				$params['replications'] = array('1'); $where_configs .= ' AND replication IN ("1")';
 				$unrestricted = TRUE;			
  			}
diff --git a/aloja-web/views/layout.html.twig b/aloja-web/views/layout.html.twig
index c16f9cb..03713eb 100644
--- a/aloja-web/views/layout.html.twig
+++ b/aloja-web/views/layout.html.twig
@@ -167,7 +167,7 @@
  		            <li {% if selected is defined and selected == 'mlparameval' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlparameval') }}">Parameter Evaluation (ML)</a></li>
 		            <li {% if selected is defined and selected == 'mloutliers' %}class="active"{% endif %}><a class="navlink" href="{{ path('mloutliers') }}">Anomaly Detection</a></li>
 		            <li {% if selected is defined and selected == 'mlminconfigs' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlminconfigs') }}">Minimal Configurations</a></li>
-		            <!--<li {% if selected is defined and selected == 'mlnewconfigs' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlnewconfigs') }}">New Cluster Configurations</a></li>-->
+		            <li {% if selected is defined and selected == 'mlnewconfigs' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlnewconfigs') }}">New Cluster Configurations</a></li>		            
 		            <li {% if selected is defined and selected == 'mlcrossvar' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlcrossvar') }}">Variable Crossing</a></li>
 		            <li {% if selected is defined and selected == 'mlcrossvar3d' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlcrossvar3d') }}">Variable Crossing (vs Exe.Time)</a></li>
 		            <li {% if selected is defined and selected == 'mlcrossvar3dfa' %}class="active"{% endif %}><a class="navlink" href="{{ path('mlcrossvar3dfa') }}">Variable Crossing (vs Prediction)</a></li>
